src/EnergyPlus/Boilers.cc

```diff
@@ -93,19 +93,19 @@ namespace EnergyPlus::Boilers {
 // METHODOLOGY EMPLOYED:
 // The BLAST/DOE-2 empirical model based on mfg. data
 
-PlantComponent *BoilerSpecs::factory(EnergyPlusData &state, std::string const &objectName)
+BoilerSpecs *BoilerSpecs::factory(EnergyPlusData &state, std::string const &objectName)
```

Comment:
If the pointer that receives this function return is declared as PlantComponent *compPtr, then what difference does it make how the factory is declared? I changed this based on a side discussion but now am thinking that either of these declarations gives the same result, compPtr is of class PlantComponent. Side discussion: if declared as PlantComponent, pointer works fine but boiler struct is inaccessable (I wanted to use for example compPtr->NomCap and that doesn't work), if declared as BoilerSpecs, the return pointer is, behind the scenes, converted to PlantComponent, and we are back to the same point. So why change the Class definition of this function?

Reply:
> If the pointer that receives this function return is declared as PlantComponent *compPtr, then what difference does it make how the factory is declared?

Right, if the pointer that receives the function is declared as PlantComponent *, then it doesn't matter.  Either way the cast is made from a Boiler * to a PlantComponent *.  If this is the only use case ever, then the factory can return either one, no difference.

But, what if you wanted to call the factory from something else.  Perhaps there is a chunk of code in E+ that loops over all boilers to report boiler usage for an energy summary.  Or perhaps there is a unit test that wants to create a single boiler and access it from the factory.  In either of those cases, having the factory return a Boiler * opens up more possibility, and you don't have to go from a Boiler * to return a PlantComponent * just to cast back to a Boiler * to use the outcome.  The factory should return the concrete type pointer and the assignment can to a cast to a base class if desired.

Reply:
This adds to the side discussion. I see your point now.